action.yml

```diff
@@ -154,6 +154,13 @@ runs:
         tutor local do importdemocourse
       shell: bash
 
+    - name: Populate Retirement States
+      run: |
+        source .tutor_venv/bin/activate
+
+        tutor local exec lms python manage.py lms populate_retirement_states
+      shell: bash
```

Comment:
I completely missed this change. Is it needed for a specific plugin?

Reply:
Yes, for the `eox-core` plugin for the support API integration tests, I explained a little more [here](https://edunext.atlassian.net/browse/AP-1527?focusedCommentId=39870). I'm not sure if we should leave this step for all cases.

Reply:
I thought about how feasible it is to add these kinds of steps right in the bash file, [here](https://github.com/eduNEXT/integration-test-in-tutor/blob/main/action.yml#L187), but I don't think it'd work since we're using a different environment to run the integration tests.

Should we take a similar approach to that but for tutor-based commands? Or we could also use tutor-contrib-picasso add extra commands utility, and allow to add extra commands through configurations.

Reply:
> I thought about how feasible it is to add these kinds of steps right in the bash file, [here](https://github.com/eduNEXT/integration-test-in-tutor/blob/main/action.yml#L187), but I don't think it'd work since we're using a different environment to run the integration tests.

I tried adding the command to the script but it doesn't find the **tutor** command.

> Should we take a similar approach to that but for tutor-based commands? Or we could also use tutor-contrib-picasso add extra commands utility, and allow to add extra commands through configurations.

I created a new step similar to the "Run integration test" step to allow passing a shell file with the Tutor extra commands. What do you think? https://github.com/eduNEXT/integration-test-in-tutor/commit/8fb3d7ab913e4884cd186a023cdc784c80538aff. In [this PR](https://github.com/eduNEXT/eox-core/actions/runs/11727125068?pr=300) was tested successfully.